src/main/java/com/google/devtools/build/lib/packages/RuleClass.java

```diff
@@ -200,6 +200,38 @@ TConfiguredTarget create(TContext ruleContext)
     public static final class RuleErrorException extends Exception {}
   }
 
+  /** Describes whether a rule implementation allows additional execution platform constraints. */
+  public enum ExecutionPlatformConstraintsAllowed {
+    /** The rule implementation does not allow any additional execution platform constraints. */
+    NONE(false, false),
+    /**
+     * The rule allows additional execution platform constraints to be added at the rule level,
```

Comment:
Since I think of a rule as an instance of a target, I find the rule / target distinction a bit confusing. Is there a less ambiguous way to describe this?

Reply:
Looks better to  me, thanks!